### PR TITLE
fix(components/accordion): allow opened panel to be closed #684 - add accordion tests

### PR DESCRIPTION
**Pull request by mufasubhai**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # 618

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Breaking changes

Please document the breaking changes if suitable.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A: tested locally
- [x] Test B: ran tests;
- [x] Test C: added new tests for space and clciok toggle

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/lib/components/Accordion/Accordion.tsx [hunk 1]
```diff
@@ -44,10 +44,19 @@ const AccordionComponent: FC<AccordionProps> = ({
   ...props
 }) => {
   const [isOpen, setOpen] = useState(collapseAll ? -1 : 0);
+
+  const toggleOpen = (i: number) => {
+    if (isOpen === i) {
```
Comment (rluders) [inline src/lib/components/Accordion/Accordion.tsx:49]:
```ts
const toggleOpen = (i: number) => setOpen(isOpen === i ? -1 : i);
```

**Review by rluders — CHANGES_REQUESTED** [review]
Don't forget to rune the `npm run prettier:fix` before commit. :)

File: src/lib/components/Accordion/Accordion.tsx (hunk 1 above)
Comment (mufasubhai) [reply src/lib/components/Accordion/Accordion.tsx]: thanks @rluders  re-ran prettier and changed toggleOpen to a ternary. Let me know anything else needs updating!

Comment (mufasubhai) [reply src/lib/components/Accordion/Accordion.tsx]: Yup. Tested locally and reran the tests using the "test:open" script.

Comment (mufasubhai) [reply src/lib/components/Accordion/Accordion.tsx]: Let me try it out. I was getting that error for a second and then I stopped getting it, but I'll check out this different syntax and see if that resolves the issue.

Appreciate the feedback!

Comment (mufasubhai) [reply src/lib/components/Accordion/Accordion.tsx]: Fix has been pushed. LMK if it looks better. Thanks for your patience!

**Review by rluders — APPROVED** [review]
Thank you @mufasubhai

**Pull request merged**